test/index.js

```diff
@@ -1 +1,138 @@
 'use strict';
+
+/* global describe, it */
+
+const assert = require('assert');
+const fs = require('fs');
+
+const R = require('ramda');
+const S = require('sanctuary');
+const $ = require('sanctuary-def');
+
+const H = require('..');
+
+const nothing = S.Nothing;
+const just = S.Just;
+
+const eq = function(actual, expected) {
+  assert.strictEqual(arguments.length, 2);
+  assert.strictEqual(R.toString(actual), R.toString(expected));
+};
+
+const GOOD_HTML = fs.readFileSync(__dirname + '/good.html', 'utf8');
+const GOOD_OBJ  = require('./good.json');
+
+const def = $.create(true, $.env);
+
+//    stripCircular :: HtmlParserNode -> HtmlParserNode
+const stripCircular = def('stripCircular', {}, [$.Any, $.Any], _node =>
+S.pipe([R.omit(['next', 'parent', 'prev']),
+        R.ifElse(R.propSatisfies(R.isNil, 'children'),
+                 R.identity,
+                 R.over(R.lensProp('children'), R.map(stripCircular)))],
+       _node));
+
+const stripNodes = S.compose(stripCircular, R.prop('value'));
+
+describe('attr', () => {
+  it('returns value of attribute for given node #1', () => {
+    const node = S.pipe([H.parse,
+                         R.nth(0)],
+                        '<h1 class="bigtitle">My text</h1>');
+    eq(H.attr('class', node), just('bigtitle'));
+  });
+  it('returns value of attribute for given node #2', () => {
+    const node = S.pipe([H.parse,
+                         R.nth(0)],
+                        '<p id="main-paragraph">What a great webpage!</p>');
+    eq(H.attr('id', node), just('main-paragraph'));
+  });
+  it('returns value of arbitrary attribute', () => {
+    const node = S.pipe([H.parse,
+                         R.nth(0)],
+                        '<p myattribute="booyah">What a great webpage!</p>');
+    eq(H.attr('myattribute', node), just('booyah'));
+  });
+  it('returns Nothing if attribute does not exist', () => {
+    const node = S.pipe([H.parse,
+                         R.nth(0)],
+                        '<p id="main-paragraph">What a great webpage!</p>');
+    eq(H.attr('style', node), nothing());
+  });
+  it('returns empty string value for boolean attributes', () => {
+    const node = S.pipe([H.parse,
+                         R.nth(0)],
+                        '<input type="checkbox" checked>');
+    eq(H.attr('checked', node), just(''));
+  });
+});
+
+describe('children', () => {
+  it('returns child nodes given parent node #1', () => {
+    const parent = S.pipe([H.parse,
+                           R.filter(H.is('html')),
+                           R.nth(0)],
+                          GOOD_HTML);
+    // Assert children html
+    eq(S.pipe([H.children, R.map(H.html), R.join('')], parent),
+       fs.readFileSync(__dirname + '/children.html', 'utf8'));
+    // Assert children object
+    eq(S.pipe([H.children, R.map(stripNodes)], parent),
+       require('./children.json'));
+  });
```

Comment:
@davidchambers I decided it was best to assert both the stripped object and the output HTML where possible, since neither perfectly capture all information about the output from something like `H.children`.